features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/MediationPolicies/Edit/EditFaultMediationPolicy.jsx

```diff
@@ -0,0 +1,376 @@
+        display: 'flex',
+        flexDirection: 'row',
+        padding: `${theme.spacing.unit * 2}px 2px`,
+    },
+    dropzone: {
+        border: '1px dashed ' + theme.palette.primary.main,
+        borderRadius: '5px',
+        cursor: 'pointer',
+        height: 'calc(100vh - 50em)',
+        padding: `${theme.spacing.unit * 2}px 0px`,
+        position: 'relative',
+        textAlign: 'center',
+        width: '80%',
+        margin: '10px 0',
+    },
+    dropZoneWrapper: {
+        height: '100%',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        '& span': {
+            fontSize: 16,
+            color: theme.palette.primary.main,
+        },
+    },
+});
+/**
+* Download the mediation policy related file
+* @param {any} response Response of download file
+*/
+const downloadFile = (response) => {
+    let fileName = '';
+    const contentDisposition = response.headers['content-disposition'];
+
+    if (contentDisposition && contentDisposition.indexOf('attachment') !== -1) {
+        const fileNameReg = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+        const matches = fileNameReg.exec(contentDisposition);
+        if (matches != null && matches[1]) fileName = matches[1].replace(/['"]/g, '');
+    }
+    const contentType = response.headers['content-type'];
+    const blob = new Blob([response.data], {
+        type: contentType,
+    });
+    if (typeof window.navigator.msSaveBlob !== 'undefined') {
+        window.navigator.msSaveBlob(blob, fileName);
+    } else {
+        const URL = window.URL || window.webkitURL;
+        const downloadUrl = URL.createObjectURL(blob);
+
+        if (fileName) {
+            const aTag = document.createElement('a');
+            if (typeof aTag.download === 'undefined') {
+                window.location = downloadUrl;
+            } else {
+                aTag.href = downloadUrl;
+                aTag.download = fileName;
+                document.body.appendChild(aTag);
+                aTag.click();
+            }
+        } else {
+            window.location = downloadUrl;
+        }
+
+        setTimeout(() => {
+            URL.revokeObjectURL(downloadUrl);
+        }, 100);
+    }
+};
+
+/**
+ * The component to manage IN mediation policies.
+ * @param {any} props The props passed to the layout
+ * @returns {any} HTML representation.
+ */
+function EditFaultMediationPolicy(props) {
+    const {
+        classes, api, updateMediationPolicy,
+    } = props;
+    const [globalFaultMediationPolicies, setGlobalFaultMediationPolicies] = useState([]);
+    // user uploaded api specific mediation policies
+    const [faultSeqCustom, setFaultSeqCustom] = useState([]);
+    const { id } = api;
+    const type = 'FAULT';
+    const selectedPolicy = api.mediationPolicies.filter(seq => seq.type === type)[0];
+    const [selectedPolicyFile, setSelectedPolicyFile] = useState({
+        id: selectedPolicy !== (null || undefined) ? selectedPolicy.id : '',
```

Comment:
Can't we just use `selectedPolicy ? selectedPolicy.id : ''` instead of checking for `null || undefined`?

Reply:
Fixed.